timescaledb/tutorials/nfl-fantasy-league.md

```diff
@@ -0,0 +1,188 @@
+
+1. [Create tables][]
+2. [Ingest data from CSV files][] 
+3. [Analyze NFL data][]
+   
+## Prerequisites
+
+* Python 3
+* TimescaleDB (see [installation options][install-timescale]) 
+* [Psql][psql-install] or any other PostgreSQL client (e.g. DBeaver)
+* Pandas and Matplotlib (`pip install pandas matplotlib`)
+
+## About the dataset
+
+The dataset we're using in this tutorial is provided by the National Football League (NFL) as part of their yearly
+Big Data Bowl. Big Data Bowl is a contest for data professionals and students to explore american football data and build interesting machine learning models.
+
+As this year's contest hasn't been published yet by the NFL, we're using last year's dataset that includes data about all the passing plays of the 2018 season.
+
+Download dataset here!
+
+## Create tables
+
+You will need to create four tables:
+
+* game
+  
+  Information about each game.
+* player
+  
+  Player information.
+* play
+  
+  Play information
+* tracking
+  
+  Tracking information from each play.
+
+```sql
+CREATE TABLE game (
+    gameId INT PRIMARY KEY,
+    gameDate DATE,
+    gameTimeEastern TIME,
+    homeTeamAbbr VARCHAR(3),
+    visitorTeamAbbr VARCHAR(3),
+    week INT
+);
+
+CREATE TABLE player (
+    nflId INT,
+    height TEXT,
+    weight INT,
+    birthDate DATE,
+    collegeName TEXT,
+    position TEXT,
+    displayName TEXT
+);
+
+CREATE TABLE play (
+    gameId INT,
+    playId INT,
+    playDescription TEXT,
+    quarter INT,
+    down INT,
+    yardsToGo INT,
+    possessionTeam TEXT,
+    playType TEXT,
+    yardlineSide TEXT,
+    yardlineNumber INT,
+    offenseFormation TEXT,
+    personnelO TEXT,
+    defendersInTheBox INT,
+    numberOfPassRushers INT,
+    personnelD TEXT,
+    typeDropback TEXT,
+    preSnapVisitorScore INT,
+    preSnapHomeScore INT,
+    gameClock TIME,
+    absoluteYardlineNumber INT,
+    penaltyCodes TEXT,
+    penaltyJerseyNumber TEXT,
+    passResult TEXT,
+    offensePlayResult INT,
+    playResult INT,
+    epa DOUBLE PRECISION,
+    isDefensivePI BOOLEAN
+);
+
+CREATE TABLE tracking (
+    time TIMESTAMP,
+    x DOUBLE PRECISION,
+    y DOUBLE PRECISION,
+    s DOUBLE PRECISION,
+    a DOUBLE PRECISION,
+    dis DOUBLE PRECISION,
+    o DOUBLE PRECISION,
+    dir DOUBLE PRECISION,
+    event TEXT,
+    nflId INT,
+    displayName TEXT,
+    jerseyNumber INT,
+    position TEXT,
+    frameId INT,
+    team TEXT,
+    gameId INT,
+    playId INT,
+    playDirection VARCHAR(5),
+    route TEXT
+);
+```
+
+## Ingest data from CSV files
+
+We have 3 separate CSV files for game, player, and play tables. For the tracking table, you will need to
+import data from 17 CSV files (1 file for each week in the season).
+
+Let's use a Python script that utilizes psycopg2's `copy_expert` function to ingest the data:
+
+```python
+import config
+import psycopg2
+
```

Comment:
should we add code comments to walk users through what each step is doing??
ex: 
"# First connect to the database location"
"# Create function to define cursor and import data"